script/checks/opcodes.ts

```diff
@@ -0,0 +1,195 @@
+import { type Hex, type PublicClient, toHex } from 'viem';
+
+type Opcode = number;
+type CallError = { details: string };
+
+export async function checkOpcodes(
+	client: PublicClient,
+): Promise<{ number: Hex; name: string; supported: boolean | string }[]> {
+	const opcodes = Array.from(Array(0xff + 1).keys());
+	const supported = await Promise.all(opcodes.map(async (opcode) => checkOpcode(opcode, client)));
+
+	const result: { number: Hex; name: string; supported: boolean | string }[] = [];
+	opcodes.forEach((opcode, index) => {
+		// For brevity, omit opcodes that are not known and not supported. All known opcodes are
+		// included. Supported but unknown opcodes are included with the name 'unknown'.
+		const shouldOmit = !knownOpcodes[opcode] && !supported[index];
+		if (!shouldOmit) {
+			result.push({
+				number: toHex(opcode, { size: 1 }),
+				name: knownOpcodes[opcode] || 'unknown',
+				supported: supported[index],
+			});
+		}
+	});
+	return result;
+}
+
+async function checkOpcode(opcode: Opcode, client: PublicClient): Promise<boolean | 'unknown'> {
+	try {
+		await client.call({ data: toHex(opcode, { size: 1 }) });
+		return true; // Call succeeded so opcode is supported.
+	} catch (e: unknown) {
+		const err = e as CallError;
+		// TODO These might be specific to the node implementation, can this be more robust?
```

Comment:
Maybe each opcode has to have an accompanying test case that succeeds when executed?

It's extra work, but much more robust. Happy to help with that if you think it's a good idea.

Reply:
I agree this is a good idea. I'm just about ready to merge this branch and hope to merge it early this week, so I've created https://github.com/mds1/evm-diff/issues/67 to track. Definitely would appreciate your help on this!